src/control/server/ctl_storage_rpc.go

```diff
@@ -151,6 +169,93 @@ func (c *ControlService) scanScm(ctx context.Context, req *ctlpb.ScanScmReq) (*c
 	return newScanScmResp(c.getScmUsage(ssr))
 }
 
+// Adjust the NVME available size to its real usable size.
+func (c *ControlService) adjustNvmeSize(resp *ctlpb.ScanNvmeResp) {
+	for _, ctl := range resp.GetCtrlrs() {
+		for _, dev := range ctl.GetSmdDevices() {
+			if dev.GetDevState() != "NORMAL" {
+				c.log.Infof("Skipping unusable device %s (%s): state=%s",
+					dev.GetUuid(), ctl.GetPciAddr(), dev.GetDevState())
+				continue
+			}
```

Comment:
At the end it is complementary with the test in DMG. If the device is defunct, its available size will not be adjusted with this test and it will not be taken into account by the DMG test.